lib/active_job/queue_adapters/resque_ext.rb

```diff
@@ -261,7 +261,7 @@ def requeue(job)
           resque_job = job.raw_data
           resque_job["retried_at"] = Time.now.strftime("%Y/%m/%d %H:%M:%S")
 
-          redis.lset(queue_redis_key, job.position, resque_job)
+          redis.lset(queue_redis_key, job.position, resque_job.to_json)
```

Comment:
```suggestion
          redis.lset(queue_redis_key, job.position, Resque.encode(resque_job))
```
I realised this is what Resque itself does in similar cases, for example [here](https://github.com/resque/resque/blob/2f9d080ce86eb2e3f1f3d47599a21c576124c6f3/lib/resque/failure/redis_multi_queue.rb#L77), which [later calls `lset` with that value](https://github.com/resque/resque/blob/2f9d080ce86eb2e3f1f3d47599a21c576124c6f3/lib/resque/data_store.rb#L204-L206).

Reply:
Yes this is in line with Resque code :+1: 
I haven't found the time to dig further, I wonder why previous versions of redis-rb support Hash and make the conversion though. In any case `Resque.encode` [specific serialization logic](https://github.com/resque/resque/blob/master/lib/resque.rb#L34-L40) so it's probably best to reuse it